hw/ip/lc_ctrl/doc/_index.md

```diff
@@ -324,17 +524,51 @@ A decoded version of this counter is exposed in the {{< regref "LC_TRANSITION_CN
 
 ### Life Cycle State Controller
 
 The life cycle state controller is the main entity that handles life cycle requests, escalation events and transactions with the OTP and flash controllers.
+The state diagram for the controller FSM is shown below.
 
-#### Life Cycle Requests
+![LC Controller FSM](lc_ctrl_fsm.svg)
+
+Once the FSM has initialized upon request from the power manager, it moves into `IdleSt`, which is the state where all life cycle control signals are broadcast.
+The life cycle controller stays in `IdleSt` unless a life cycle state request is initiated via the CSRs.
+
+In that case, the life cycle controller first increments the redundantly encoded life cycle transition counter in `CntIncrSt` and `CntProgSt` in order to fend against brute force attacks.
+Then, the transition is checked for validity in `TransCheckSt` and the token hashing operation is initiated in `TokenHashSt`.
+A first token comparison is performed when the hashed token returns in `TokenHashSt`, followed by two more comparisons in `TokenCheck0St` and `TokenCheck1St`.
+The difference among these three comparisons is that the first comparison is done using the hashed token input directly, whereas the second and the third comparison use a registered version of the hashed token.
+If all token checks are successful, the next life cycle state vector is computed and programmed in `TransProgSt`.
+
+Note that an initiated life cycle transition request always ends in `PostTransSt`, no matter whether the transition is successful or not.
+
+#### Escalation Handling
+
+The life cycle controller contains two escalation channels that are connected to the [alert handler]({{< relref "hw/ip/alert_handler/doc" >}}).
+
+When the first channel `esc_wipe_secrets` is asserted, the life cycle controller permanently asserts the `lc_escalate_en` life cycle signal.
+That signal is routed to various security modules in OpenTitan and triggers local wiping and invalidation features.
+Note that this first escalation action does not affect the life cycle state.
+
+When the second channel `esc_scrap_state` is asserted, the life cycle controller moves the life cycle state into `EscalateSt`, which behaves like a "virtual" SCRAP life cycle state.
+This transition is not permanent, and will clear upon the next power cycle.
+
+#### FSM Glitch Countermeasures
+
+The FSM has been designed to have a linear control flow that always moves in the same direction, and that always ends in a terminal state after initiating a transition request in order to make glitch attacks harder.
+A sparse FSM state encoding is employed, where each state is encoded as a 16bit word with a minimum Hamming distance of 5 w.r.t. any other state.
+The FSM state and the life cycle state vector are concurrently monitored, and if an erroneous encoding is detected, the life cycle FSM is immediately moved into the terminal `InvalidSt`, and a `lc_state_failure` alert is asserted.
```

Comment:
does the `InvalidSt` also correspond to a virtual scrap? Or would that need to take the alert loop?

Reply:
yes, and no this does not have to go via the alert loop.

we have a couple of states that are all equivalent to virtual scrap, with the only differences that they have a different encoding, and that we arrive in these states via different avenues.

I.e., `InvalidSt`, `PostTransitionSt` and `EscalateSt` all behave like virtual scrap, but they have different causes:
- `InvalidSt`: any error in either the life cycle state vector or the redundantly encoded FSM will immediately move the FSM into this state - no escalation required.
- `PostTransitionSt`: as the name says, every transition request eventually ends in this state
- `EscalateSt`: we will move here via the alert handler escalation path.